mongo/mongo.go

```diff
@@ -153,21 +154,32 @@ func (m *Mongo) ListInteractives(ctx context.Context, offset, limit int, modelFi
 // generate filter string depending on data type
 // string eq value
 // array in value(s)
-func generateFilter(model *models.InteractiveMetadata) bson.M {
+func generateFilter(model *models.InteractiveFilter) bson.M {
 	filter := bson.M{}
 	filter["active"] = bson.M{"$eq": true}
-	if model == nil {
+	if model == nil || model.Metadata == nil {
+		return filter
+	}
+
+	// if filter by collection-id
```

Comment:
dunno if we need these comments - the code enough imo 😊  but upto you

Reply:
can i leave them ? :) .. will forget what/why i did in a week